spec/lib/two_factor_authentication/models/two_factor_authenticatable_spec.rb

```diff
@@ -90,6 +90,12 @@ def do_invoke(code, user)
         code = @totp_helper.totp_code(1.minutes.ago.to_i)
         expect(do_invoke(code, instance)).to eq(false)
       end
+
+      it 'prevents code reuse' do
+        code = @totp_helper.totp_code
+        expect(do_invoke(code, instance)).to eq(true)
+        expect(do_invoke(code, instance)).to eq(false)
+      end
```

Comment:
As per our offline discussion this kind of advanced testing probably belongs in the rotp gem itself.  I will follow up there.